File: 04_Currency_Mocks/java/01_start/src/main/java/CurrencyConverter.java
public class CurrencyConverter {

    private final ExchangeRateService changeRates;

    public CurrencyConverter(ExchangeRateService changeRates) {
        this.changeRates = changeRates;
    }

    public Double convert(String fromCurrency, String toCurrency, Double amount) {
        throw new UnsupportedOperationException("Unimplemented method");
    }
    
}

File: 05_Currency_Mocks/ts/01_start/src/currency-converter.ts
import { ExchangeRateService } from "./exchange-rate-service";

export class CurrencyConverter {
  exchangeRates: ExchangeRateService;

  constructor(exchangeRates: ExchangeRateService) {
    this.exchangeRates = exchangeRates;
  }

  convert(from: string, to: string, amount: number) {
    throw new Error("not implemented");
  }
}
